### PR TITLE
update to protobuf 3

### PR DESCRIPTION
**Pull request by BusyJay**
The protocol of Merge bases on protobuf 3, so let's upgrade it.

See also pingcap/kvproto#208

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -919,6 +872,53 @@ dependencies = [
 ]
 
 [[package]]
+name = "tikv"
```
Comment (overvenus) [inline Cargo.lock]: Why did it move here?

Comment (BusyJay) [reply Cargo.lock]: Latest rls moves it. And it should be compatible with old compiler.

**Comment by disksing** [comment]
LGTM.

**Comment by BusyJay** [comment]
PTAL

**Comment by BusyJay** [comment]
ping

**Comment by zhangjinpeng1987** [comment]
@overvenus

**Pull request merged**